### PR TITLE
Add server request logging

### PR DESCRIPTION
**Pull request by adamstruck**

### CODE REVIEW

**Comment by kellrott** [comment]
After fixing the bug that the patch seems to be causing, the verbosity of the logging should probably be turned down during testing.

File: server/config.go [hunk 1]
```diff
@@ -26,6 +26,12 @@ type Config struct {
 	// Strategy to use for selecting the vertices/edges to inspect.
 	// Random if True; first N otherwise
 	SchemaRandomSample bool
+	// Configure how the server logs requests
+	RequestLogging struct {
+		Disable bool
```
Comment (kellrott) [inline server/config.go]: Maybe switch this to Enable, so the default value (false) means no request logging takes place.

File: server/logging_middleware.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+			"path":    info.FullMethod,
+			"request": req,
+			"headers": headers,
+			"latency": time.Since(start).String(),
+			"status":  runtime.HTTPStatusFromCode(status.Code(err)),
+		})
+		if err == nil {
+			entry.Info("gRPC server responded")
+		} else {
+			entry.Error("gRPC server responded")
```
Comment (kellrott) [inline server/logging_middleware.go]: Are both paths supposed to have the same message? Maybe log the contents of err?

File: server/server.go [hunk 3]
```diff
@@ -206,7 +146,38 @@ func (server *GripServer) Serve(pctx context.Context) error {
...
+				"path":    req.URL.Path,
+				"request": string(body),
+				"header":  headers,
+				"latency": time.Since(start).String(),
+				"status":  lrw.statusCode,
+			})
+			if lrw.statusCode == http.StatusOK {
+				entry.Info("HTTP server responded")
+			} else {
+				entry.Error("HTTP server responded")
```
Comment (kellrott) [inline server/server.go]: Log the non ok status code?

Comment (adamstruck) [reply server/server.go]: The status code is always logged. This logic just sets the log level.

File: server/logging_middleware.go (hunk 2 above)
Comment (adamstruck) [reply server/logging_middleware.go]: Yeah the only difference was supposed to be the log level. I can add the error message, the reason I didn’t initially was to make these logs as similar to the ones from the http server as possible.

File: server/config.go (hunk 1 above)
Comment (adamstruck) [reply server/config.go]: Will do.

**Comment by kellrott** [comment]
👍

**Pull request merged**